### PR TITLE
go sdk transformer

### PR DESCRIPTION
**Pull request by starkballc**
# Updates
- [ TSP-6174 ] Implement transformation logic in go sdk

### Important Notes
- implemented transformation logic and tested in go sdk
- created Route struct
- lack of coverage due to old code

### CODE REVIEW

**Review by starkshears — DISMISSED** [review]

**Review by starkshears — DISMISSED** [review]

File: pkg/domain/transformer.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package domain
+
+import (
+	logger "github.com/sirupsen/logrus"
+	"strconv"
+	"strings"
+)
+
+func getTransMap(s string) (map[interface{}]interface{}, error) {
```
Comment (starkmichelsc) [inline pkg/domain/transformer.go]: the return should be a float[float] as you are parsing to a float. this well prevent a need for further casting in follow up code

File: pkg/domain/msgResp.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package domain
+
+type MsgResp struct {
```
Comment (starkmichelsc) [inline pkg/domain/msgResp.go]: what is the purpsoe of this struct? It looks like the only field that is used is the value. Is this struct needed or are there plans for using the other values in later changes?

File: pkg/domain/telemetry-message.go [hunk 3]
```diff
@@ -13,33 +13,33 @@ const (
...
-	values   map[string]float64
+	Values   map[string]float64
 }
 
 func NewTelemetryMessage(deviceId string) *TelemetryMessage {
 	n := new(TelemetryMessage)
 	n.Ts = time.Now().Unix()
 	n.DeviceId = deviceId
-	n.values = make(map[string]float64)
+	n.Values = make(map[string]float64)
```
Comment (starkmichelsc) [inline pkg/domain/telemetry-message.go]: values can be private given the provided Set/Get funcs

File: pkg/domain/telemetry-message_test.go [hunk 4]
```diff
@@ -35,7 +35,7 @@ func TestTelemetryMessage_SetValue(t *testing.T) {
 
 	message.SetValue(key, value)
 
-	assert.Equal(t, value, message.values[key])
+	assert.Equal(t, value, message.Values[key])
```
Comment (starkmichelsc) [inline pkg/domain/telemetry-message_test.go]: Update to use the Getter

**Review by starkmichelsc — CHANGES_REQUESTED** [review]

File: pkg/domain/transformer.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
+package domain
+
+import (
+	logger "github.com/sirupsen/logrus"
+	"strconv"
+	"strings"
+)
+
+func getTransMap(s string) (map[float64]float64, error) {
```
Comment (starkmichelsc) [inline pkg/domain/transformer.go]: rename to parseTransMap

File: pkg/domain/route.go [hunk 6]
```diff
@@ -0,0 +1,6 @@
+package domain
+
+type Route struct {
```
Comment (starkmichelsc) [inline pkg/domain/route.go]: please rename to something that does not conflict with route as that is used for handlers/endpoints. maybe TransformerValue? and add the float64 to this struct as you will need to convert from a iot-message-bus.Route -> this struct to use the sdk. This struct would also be used in TDS to revert to the raw value as well. "Route" is specific to iot-message-bus

**Review by starkmichelsc — CHANGES_REQUESTED** [review]

File: pkg/domain/transformer.go [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+				return nil, errVal
+			}
+
+			m[key] = value
+		}
+	}
+	return m, nil
+}
+
+func Transform(telem *TelemetryMessage, route *TransformerValue) (*TelemetryMessage, error) {
```
Comment (starkmichelsc) [inline pkg/domain/transformer.go]: this takes in a telem pointer but also returns a telem pointer. it should either
1. take in a point and only return error( alter inline)  
2. take in a non-pointer and return a non-pointer(alter and return)

rename route param. 
Does the router param need to be a pointer?

**Review by starkmichelsc — CHANGES_REQUESTED** [review]

**Review by starkmichelsc — APPROVED** [review]

**Review by starkshears — APPROVED** [review]

**Pull request merged**